src/llama_stack_client/lib/agents/agent.py

```diff
@@ -49,40 +71,27 @@ async def create_turn(
             attachments=attachments,
             stream=True,
         )
-        turn = None
         for chunk in response:
-            if chunk.event.payload.event_type != "turn_complete":
+            if not self._has_tool_call(chunk):
                 yield chunk
-            else:
-                turn = chunk.event.payload.turn
-
-        message = turn.output_message
-        if len(message.tool_calls) == 0:
-            yield chunk
-            return
-
-        if message.stop_reason == "out_of_tokens":
-            yield chunk
-            return
-
-        tool_call = message.tool_calls[0]
-        if tool_call.tool_name not in self.custom_tools:
-            m = ToolResponseMessage(
-                call_id=tool_call.call_id,
-                tool_name=tool_call.tool_name,
-                content=f"Unknown tool `{tool_call.tool_name}` was called. Try again with something else",
-                role="ipython",
-            )
-            next_message = m
-        else:
-            tool = self.custom_tools[tool_call.tool_name]
-            result_messages = await self.execute_custom_tool(tool, message)
-            next_message = result_messages[0]
-
-        yield next_message
 
-    async def execute_custom_tool(
-        self, tool: CustomTool, message: Union[UserMessage, ToolResponseMessage]
-    ) -> List[Union[UserMessage, ToolResponseMessage]]:
-        result_messages = await tool.run([message])
-        return result_messages
+    async def async_create_turn(
+        self,
+        messages: List[Union[UserMessage, ToolResponseMessage]],
+        attachments: Optional[List[Attachment]] = None,
+        session_id: Optional[str] = None,
+    ):
+        response = self.client.agents.turn.create(
+            agent_id=self.agent_id,
+            # use specified session_id or last session created
+            session_id=session_id or self.session_id[-1],
+            messages=messages,
+            attachments=attachments,
+            stream=True,
+        )
+        for chunk in response:
+            if not self._has_tool_call(chunk):
+                yield chunk
+            else:
+                next_message = await self._async_run_tool(chunk)
+                yield next_message
```

Comment:
something to fix for us later, but it is kind of weird how we don't send this message out as a chunk also.

Reply:
Yeah, this message is from client side's CustomTool, while `AgentTurnResponseStreamChunk` are from server.